Atomic/stop.py

```diff
@@ -16,38 +16,33 @@ def cli(subparser):
         "image does not specify LABEL STOP")
     stopp.set_defaults(_class=Stop, func='stop')
     util.add_opt(stopp)
-    stopp.add_argument("-n", "--name", dest="name", default=None,
-                       help=_("name of container"))
-    stopp.add_argument("image", help=_("container image"))
+    stopp.add_argument("container", help=_("container name or ID"))
     stopp.add_argument("args", nargs=argparse.REMAINDER,
                           help=_("Additional arguments appended to the image "
                                  "stop method"))
 
+ATOMIC_CONFIG = util.get_atomic_config()
+storage = ATOMIC_CONFIG.get('default_storage', "docker")
+
 class Stop(Atomic):
     def __init__(self):
         super(Stop, self).__init__()
 
     def stop(self):
-        if self.syscontainers.get_checkout(self.name) is not None:
-            self.syscontainers.stop_service(self.name)
-            return
-
-        self.inspect = self._inspect_container()
-        if self.inspect is None:
-            self.inspect = self._inspect_image()
-            if self.inspect is None:
-                raise DockerObjectNotFound(self.name)
-
-        args = self._get_args("STOP")
-        if args:
-            cmd = self.gen_cmd(args + self.quote(self.args.args))
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        beu = BackendUtils()
+        be, con_obj = beu.get_backend_and_container_obj(self.args.image, storage)
+        label_stop_args = con_obj.get_label('stop')
+        if label_stop_args and be.backend == 'docker':
```

Comment:
Long term it should be ... but this would require splicing and dicing the Atomic class because we use two methods in Atomic to format out args.  I'm not ready to dice that up yet.  Think we can forego it for now?